group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -360,14 +334,13 @@ public CompletableFuture<JoinGroupResponseData> joinGroup(
             Duration.ofMillis(config.offsetCommitTimeoutMs),
             coordinator -> coordinator.classicGroupJoin(context, request, responseFuture)
         ).exceptionally(exception -> {
-            if (!(exception instanceof KafkaException)) {
```

Comment:
There are a few of these requests that handled it like this but now we will return specific errors and not say "unexpected exception"

Is it the case that these were never really unexpected (besides unknown server error)
Do we want to flag any other errors as unexpected? Or just handle any of them/return in the response without a log.

Reply:
Do we also want to add tests for some of the new error handling?

Reply:
It will be handled by https://github.com/apache/kafka/pull/15196/files#diff-ea6dfd70c68aa99b5dc43eecadb8f3b555cb5a54162cd681bcc1477fd936913dR1080. The non KafkaExceptions are converted to UNKNOWN_SERVER_ERROR and are logged.

The change relies on the existing tests. The overall error handling did not change.

Reply:
Hmm -- in the KafkaException case we are converting errors though -- so the error handling for those seems to be changing right? if we return unknown_topic_or_partition we will convert the response to coordinator not available. 

Are you saying the error handling doesn't change because all the other errors are not expected to be returned and if they somehow do and get converted then it doesn't matter?

Reply:
Sorry, I did not get your point initially. The errors for those APIs are also converted but they are within ´classicGroupJoin’ based on the same scheme. So doing it here does not hurt. As a follow up, I’d like to remove the translation there in order to have it centralized here.